developer/reference/articles/tooling/blade-cli/08-extending-blade-cli/05-creating-a-blade-profile.markdown

```diff
@@ -0,0 +1,113 @@
+
+## Creating a New Profile
+
+To create a new Blade profile, follow these steps:
+
+1.  [Create a generic OSGi module](/docs/7-2/reference/-/knowledge_base/r/creating-a-project).
+
+2.  To create a new command, create the command and arguments classes extending
+    [`BaseCommand`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseCommand.java)
+    and
+    [`BaseArgs`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseArgs.java),
+    respectively, as described in the
+    [Creating Custom Commands](/docs/7-2/reference/-/knowledge_base/r/creating-custom-commands-for-blade-cli)
+    article. These classes should reside in the profile module's
+    `src/main/java/PACKAGE_NAME` folder. These classes register your command and
+    arguments to Blade CLI.
+
+3.  To override a default command, create new command and arguments classes
+    similar to what was done
+    [here](/docs/7-2/reference/-/knowledge_base/r/creating-custom-commands-for-blade-cli).
+    Instead of extending the
+    [`BaseCommand`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseCommand.java)
+    and
+    [`BaseArgs`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BaseArgs.java),
+    classes, extend the command/arguments classes defined for the command you
+    intend to override. Make sure to also set the `@Parameters` annotation's
+    `commandNames` argument to the command to override.
+
+    For example, if you intend to override the default `deploy` command, your
+    arguments class declaration would look like this:
+
+    ```java
+    @Parameters(commandDescription = "Overridden Deploy Command", commandNames = "deploy")
+    public class OverriddenArgs extends DeployArgs {
+
+    }
+    ```
+
+    You can search for the default command/arguments classes
+    [here](https://github.com/liferay/liferay-blade-cli/tree/master/cli/src/main/java/com/liferay/blade/cli/command).
+
+4.  To associate a command to your new profile, set the
+    [`BladeProfile`](https://github.com/liferay/liferay-blade-cli/blob/master/cli/src/main/java/com/liferay/blade/cli/command/BladeProfile.java)
+    annotation in your command class:
+
+    ```java
+    @BladeProfile("myprofile")
+    public class NewCommand extends BaseCommand<NewArgs> {
+
+    }
+    ```
+
+    The annotation's parameter should specify the profile you want to associate
+    the command with (e.g., `myprofile`).
+
+Excellent! You've created a new Blade profile and learned how to add new
+commands or override default commands by leveraging the profile. You can
+reference the
+[sample profile project](https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-profile)
+to examine a new command and overridden command's setup in a custom profile.
+
```

Comment:
Should we remind the reader about the `src/main/resources/META-INF/services/com.liferay.blade.cli.command.BaseCommand` file here?